common/types.go

```diff
@@ -35,7 +35,10 @@ const (
 var hashJsonLengthErr = errors.New("common: unmarshalJSON failed: hash must be exactly 32 bytes")
 
 type (
-	Hash    [HashLength]byte
+	// Hash represents the 32 byte Keccak256 hash of arbitrary data.
+	Hash [HashLength]byte
```

Comment:
common.Hash doesn't always represent keccak256 hashes. The value can be the result of any hash function with 256 bits of output.